ft-sdk/src/session/mod.rs

```diff
@@ -64,28 +83,15 @@ impl SessionID {
             .filter(fastn_session::id.eq(self.0.as_str()))
             .first::<String>(conn)?;
 
-        Ok(serde_json::from_str(&data).expect("session data must be serializable json object"))
-    }
+        let data: std::collections::HashMap<String, serde_json::Value> =
+            serde_json::from_str(&data).expect("session data must be serializable json object");
 
-    pub fn get_key<S: AsRef<str>, V: serde::de::DeserializeOwned>(
-        &self,
-        conn: &mut ft_sdk::Connection,
-        k: S,
-    ) -> Result<V, GetKeyError> {
-        let data = self.data(conn).map_err(GetKeyError::DatabaseError)?;
-
-        let data = data
-            .as_object()
-            .expect("session data must be a serializable json object");
-
-        let value = data
-            .get(k.as_ref())
-            .ok_or_else(|| GetKeyError::KeyNotFound(k.as_ref().to_string()))?
-            .clone();
-
-        serde_json::from_value(value).map_err(GetKeyError::SerdeError)
+        Ok(SessionData::new(self.0.as_str(), data))
     }
 
+    /// Directly store a key-value in the session store. This will overwrite the existing value for
+    /// the given key if it exists.
+    /// This is useful for storing a single key-value pair without fetching the entire session data
```

Comment:
This function is doing a `format!()` to construct query, with data which may come from end user, and therefore is susceptible for SQL injection attack.

Reply:
Thanks! fixed in https://github.com/FifthTry/ft-sdk/pull/52/commits/4f9b77023e89ba8abd99f0f745ec9deac829f43e